core/templates/components/common-layout-directives/common-elements/promo-bar.component.ts

```diff
@@ -0,0 +1,87 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for a promo bar that appears at the top of the
+ * screen. The bar is configurable with a message and whether the promo is
+ * dismissible.
+ */
+
+import { Component, OnInit } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+
+import { PromoBarBackendApiService } from
+  'services/promo-bar-backend-api.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+
+@Component({
+  selector: 'oppia-promo-bar',
+  templateUrl: './promo-bar.component.html'
+})
+export class PromoBarComponent implements OnInit {
+  promoIsVisible: boolean;
+  promoBarIsEnabled: boolean;
+  promoBarMessage: string;
+
+  constructor(
+    private promoBarBackendApiService: PromoBarBackendApiService,
+    private windowRef: WindowRef
+  ) {}
+
+  ngOnInit(): void {
+    this.promoBarBackendApiService.getPromoBarDataAsync()
+      .then((promoBar) => {
+        this.promoBarIsEnabled = promoBar.isPromoBarEnabled();
+        this.promoBarMessage = promoBar.getPromoBarMessage();
+      });
+    this.promoIsVisible = !this.isPromoDismissed();
+  }
+
+  isPromoDismissed(): boolean {
+    if (!this.isSessionStorageAvailable()) {
+      return false;
+    }
+    return angular.fromJson(
+      this.windowRef.nativeWindow.sessionStorage.promoIsDismissed);
+  }
+
+  setPromoDismissed(promoIsDismissed: boolean): boolean {
+    if (!this.isSessionStorageAvailable()) {
+      return false;
+    }
+    this.windowRef.nativeWindow.sessionStorage.promoIsDismissed =
+      angular.toJson(promoIsDismissed);
```

Comment:
Is it necessary to convert boolean to JSON(string) here?
any thoughts? @nithusha21 @srijanreddy98

Reply:
I don't think, it is a good idea to use angular js functions in a migrated component. It might work now, but what are we going to do when we complete migration and start removing angular js completely.
Find a way which does not depend on angularjs, maybe try `JSON.stringify(promoIsDismissed) `

Reply:
@Radesh-kumar Please use JSON.stringify here.

Reply:
done!